### DISCUSSION TITLE
Announcing BrowserQL, our next-gen automation tooling

### DISCUSSION THREAD

**Discussion by zach-browserless** [browserless/chrome#4413] [Announcements]
We’re very excited to announce that BrowserQL is now live! It’s our fully revamped approach to browser automation.

![image](https://github.com/user-attachments/assets/149bc3b0-c44d-41e2-83db-c21a4db5a847)

It comes from our frustrations with Playwright and Puppeteer, which aren’t designed to be stealthy. We found ourselves constantly having to modify their behavior to hide signs of automation.We ended up creating our own internal tools to directly interact with the browser via Chrome DevTools Protocol. As is often the way, those tools have morphed into a full standalone product.The result is BrowserQL, a dedicated query language and scraping IDE. As a quick summary:

It comes from our frustrations with Playwright and Puppeteer, which aren’t designed to be stealthy. We found ourselves constantly having to modify their behavior to hide signs of automation.

We ended up creating our own internal tools to directly interact with the browser via Chrome DevTools Protocol. As is often the way, those tools have morphed into a full standalone product.

The result is BrowserQL, a dedicated query language and scraping IDE. As a quick summary

- It can get past almost all bot detectors, and our team can help with ones that need extra attention
- The query language makes helpful assumptions, such as waiting for any selectors 
- You get a live view of the browsers running in the cloud for easier debugging
- Inbuilt docs mean less window hopping and inevitable distractions
- We handle details like queueing and concurrencies as usual
- Many sites don’t even require a proxy with BQL, but it’s available as a quick toggle

For the full details, check out the announcement article here:

[Announcing BrowserQL, our next-gen scraping tool](https://aplk.red/r?p=abteheg4qp3rymrjq4abazzwesdojwzp2d4sojqlaeibb5xqkd434elmj2matz2dzvykz2feef7ot4mok6rwklb6g5nicavtc4oiuz7nvaydkmqzadgnyegwqylfqtwyox22556sqz5r54dwwsf3bp25e6ra)

As much as we'd love to make it open source, we don't want to make our stealth strategies easily available for the devs at Cloudflare/Datadome to analyse, so it is going to stay closed-source until further notice. To test it out, you'll need to [grab a trial](https://account.browserless.io/signup/email/).

**Comment by akamick86** [comment]
Did you consider adding an SDK for BrowserQL? I checked the website and it seems like the anticipated usage is to send strings. Which makes it harder to manage in case of any logic.
I'm just reading docs at: https://docs.browserless.io/browserql/connecting-libraries/puppeteer
